packages/superset-ui-core/src/query/types/AdvancedAnalytics.ts

```diff
@@ -0,0 +1,67 @@
+/* eslint-disable camelcase */
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+export enum RollingType {
+  None = 'None',
+  Mean = 'mean',
+  Sum = 'sum',
+  Std = 'std',
+  Cumsum = 'cumsum',
+}
+export interface RollingWindow {
+  rolling_type?: RollingType;
+  rolling_periods?: number;
+  min_periods?: number;
+}
+
+export type TimeShiftType =
+  | '1 day ago'
+  | '1 week ago'
+  | '28 days ago'
+  | '30 days ago'
+  | '52 weeks ago'
+  | '1 year ago'
+  | '104 weeks ago'
+  | '2 years ago';
+export enum ComparisionType {
```

Comment:
String literals might be good enough. You can use `as ComparisionType` or `const variable: ComparisionType = ...` when you need to enforce the type.

Reply:
In order to use `Object.values(ComparisionType)` to get array of `ComparisionType`.